#! /bin/sh -e

# DP: Remove duplicate PyCallable_Check declaration.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Include/abstract.h~	2004-03-12 17:38:17.000000000 +0100
+++ Include/abstract.h	2006-09-22 07:42:19.282532734 +0200
@@ -283,9 +283,10 @@
 
        */
 
+       /* Declared elsewhere
+
      PyAPI_FUNC(int) PyCallable_Check(PyObject *o);
 
-       /*
 	 Determine if the object, o, is callable.  Return 1 if the
 	 object is callable and 0 otherwise.
 
